src/internet/core/internetmodel.cpp

```diff
@@ -338,6 +340,10 @@ QMimeData* InternetModel::mimeData(const QModelIndexList& indexes) const {
 
   if (urls.isEmpty()) return nullptr;
 
+  for (const QModelIndex& index : new_indexes) {
+    InternetModel::ServiceForIndex(index)->ItemNowPlaying(itemFromIndex(index));
+  }
+
```

Comment:
The mimeData is method is just meant return info on select items. This doesn't necessarily mean that items are playing. We should look into a path from the player when the playback starts. The Player passes a MediaPlaybackRequest to the engine and the engine. That could be updated to carry the request URL. I'll try to put together a proof of concept.

Reply:
I opened request #6946 to fix the URL emitted in the SongChangeRequestProcessed signal. However, I noticed that you're no longer using a unique URI scheme and a UrlHandler for translation.

Reply:
I will reintroduce it and try to rework the code in order to use that signal

Reply:
I reworked the code to reintroduce an url handler for the "radiobrowser:" protocol.
When the url handler receives a request to resolve an url using this protocol, it calls the radiobrowser API to get the station click count increased; the API reply includes the real station url, and this is passed to the player.
I removed the dirty hack in InternetModel.
It looks like #6946 is not even needed using this approach.

Reply:
Yep, that should work for the request, and should be correct assuming that we resolve the URL once per play (which I think is the correct assumption). The other signal, if you chose to use that, would tell you that the stream successfully started playing.